pkg/eval/glob.go

```diff
@@ -75,6 +89,24 @@ func (gp GlobPattern) Index(k interface{}) (interface{}, error) {
 		gp.Segments[len(gp.Segments)-1] = glob.Wild{
 			lastSeg.Type, true, lastSeg.Matchers,
 		}
+	case modifier == "d" || modifier == "r" || strings.HasPrefix(modifier, "type:"):
```

Comment:
`type:dir` and `type:regular` and nothing more; remove "d" and "r".

Reply:
Okay. Done. That makes this more painful to use interactively but presumably you would prefer not to institute these shorthands until more experience using this feature is gained and there has been more discussion about whether shorthands like `/` for `type:dir` is preferable to `d` or some other shorthand.